networkit/csbridge.py

```diff
@@ -0,0 +1,141 @@
+# local imports
+from .support import MissingDependencyError
+from .structures import Partition
+
+# external imports
+try:
+	import ipycytoscape
+except ImportError:
+	have_cyto = False
+else:
+	have_cyto = True
+
+try:
+	import seaborn
+except ImportError:
+	have_seaborn = False
+else:
+	have_seaborn = True
+
+def widget_from_graph(G, node_scores = None, node_partition = None, node_colors = None, show_ids = True):
+	""" 
+	Creates a ipycytoscape-widget from a given graph. The returned widget already contains
+	all nodes and edges from the graph. The graph is colored using an array of norm. rgb-values
+	based on seaborn perceptually uniform color map (rocket) or a user given custom color array. 
+	See matplotlib color maps for the correct formatting: 
+	https://matplotlib.org/api/_as_gen/matplotlib.colors.Colormap.html#matplotlib.colors.Colormap 
+
+	Parameters:
+	-----------
+	G : networkit.graph.Graph
+		Graph (nodes, edges), which should be visualized
+	node_scores : list of numbers
+		List of scores for each nodes, for example scores from a centrality measure. This is 
+		used for color-calculation of the nodes (continuous distribution). Provide either 
+		node_scores or node_partition - not both.
+	node_partition : networkit.structures.Partition
+		Partition object. This is used for color-calculation of the nodes (discrete distribution). 
+		Provide either node_scores or node_partition - not both. 	
+	node_colors : list of tuples
+		Array consisting of normalized rgb-values. If none is given, seaborn.color_palette.colors is used
+	show_ids : 	boolean	
+		Set whether node ids should be visible in plot-widget. Is set to True by default. 
+	"""
+	# Sanity checks
+	if not have_cyto:
+		raise MissingDependencyError("ipycytoscape")
+
+	if node_scores is not None and node_partition is not None:
+		raise InputError("Provide either node_scores or node_partition - not both")
```

Comment:
But providing neither is OK?

Reply:
At least that was my intuition. If none is provided (last else case), a default color for all nodes is choosen. So you get a widget with uniform coloring in the end.

Reply:
Sounds reasonable.